common/strings.c

```diff
@@ -88,6 +88,91 @@ TDNFSafeAllocateString(
     goto cleanup;
 }
 
+size_t
```

Comment:
Common style in tdnf code is to return dwError, and return the value in an argument.

Reply:
This method has no errors. An error here is application/usage dependant. For instance someone can pass a string and  check if it does not have any of the give separator chars by check `count == 1`.

The mandate of using `BAIL_ON_TDNF_ERROR` is causing methods that don't need to return anything also return a `uint32_t` just so the caller can also call `BAIL_ON_TDNF_ERROR`. It's fast becoming an anti-pattern IMO. But do let me know if this is a "must be followed".

Reply:
I don't really like his, because it's different to the behaviors of similar functions and therefore maybe confusing. Caller may not know that this function cannot fail, and maybe this will change some time in the future.

I won't insist on the change.

Reply:
Changed.

A caller _must_ know know a function's contract before invoking it.